components/webdriver_server/keys.rs

```diff
@@ -2,184 +2,105 @@
-        '}' => Some((Key::RightBracket, true)),
-        ']' => Some((Key::RightBracket, false)),
-        '\u{E000}' => None,
-        '\u{E001}' => None,
-        '\u{E002}' => None,
-        '\u{E003}' => Some((Key::Backspace, false)),
-        '\u{E004}' => Some((Key::Tab, false)),
-        '\u{E005}' => None,
-        '\u{E006}' => Some((Key::Enter, false)), // This is supposed to be the Return key
-        '\u{E007}' => Some((Key::Enter, false)),
-        '\u{E008}' => Some((Key::LeftShift, false)),
-        '\u{E009}' => Some((Key::LeftShift, false)),
-        '\u{E00A}' => Some((Key::LeftAlt, false)),
-        '\u{E00B}' => Some((Key::Pause, false)),
-        '\u{E00C}' => Some((Key::Escape, false)),
-        '\u{E00D}' => Some((Key::Space, false)),
-        '\u{E00E}' => Some((Key::PageUp, false)),
-        '\u{E00F}' => Some((Key::PageDown, false)),
-        '\u{E010}' => Some((Key::End, false)),
-        '\u{E011}' => Some((Key::Home, false)),
-        '\u{E012}' => Some((Key::Right, false)),
-        '\u{E013}' => Some((Key::Left, false)),
-        '\u{E014}' => Some((Key::Down, false)),
-        '\u{E015}' => Some((Key::Up, false)),
-        '\u{E016}' => Some((Key::Insert, false)),
-        '\u{E017}' => Some((Key::Delete, false)),
-        '\u{E018}' => Some((Key::Semicolon, false)),
-        '\u{E019}' => Some((Key::Equal, false)),
-        '\u{E01A}' => Some((Key::Kp0, false)),
-        '\u{E01B}' => Some((Key::Kp1, false)),
-        '\u{E01C}' => Some((Key::Kp2, false)),
-        '\u{E01D}' => Some((Key::Kp3, false)),
-        '\u{E01E}' => Some((Key::Kp4, false)),
-        '\u{E01F}' => Some((Key::Kp5, false)),
-        '\u{E020}' => Some((Key::Kp6, false)),
-        '\u{E021}' => Some((Key::Kp7, false)),
-        '\u{E022}' => Some((Key::Kp8, false)),
-        '\u{E023}' => Some((Key::Kp9, false)),
-        '\u{E024}' => Some((Key::KpMultiply, false)),
-        '\u{E025}' => Some((Key::KpAdd, false)),
-        '\u{E026}' => Some((Key::KpEnter, false)),
-        '\u{E027}' => Some((Key::KpSubtract, false)),
-        '\u{E028}' => Some((Key::KpDecimal, false)),
-        '\u{E029}' => Some((Key::KpDivide, false)),
-        '\u{E031}' => Some((Key::F1, false)),
-        '\u{E032}' => Some((Key::F2, false)),
-        '\u{E033}' => Some((Key::F3, false)),
-        '\u{E034}' => Some((Key::F4, false)),
-        '\u{E035}' => Some((Key::F5, false)),
-        '\u{E036}' => Some((Key::F6, false)),
-        '\u{E037}' => Some((Key::F7, false)),
-        '\u{E038}' => Some((Key::F8, false)),
-        '\u{E039}' => Some((Key::F9, false)),
-        '\u{E03A}' => Some((Key::F10, false)),
-        '\u{E03B}' => Some((Key::F11, false)),
-        '\u{E03C}' => Some((Key::F12, false)),
-        '\u{E03D}' => None,
-        '\u{E040}' => None,
-        _ => None,
+// spec: https://w3c.github.io/webdriver/#keyboard-actions
+// normalised (sic) as in british spelling
+fn get_normalised_key_value(key: char) -> Key {
```

Comment:
Could that be an utility function part of keyboard-types?

Reply:
This is a webdriver specific function. If you want to include webdriver specific functions in keyboard-types I'd directly put `keycodes_to_keys` in keyboard-types which converts a string of input characters into a sequence of keydown and keyup events.